common/azure_auth.go

```diff
@@ -292,6 +317,10 @@ func (aa *AzureAuth) getClientSecretAuthorizer(resource string) (autorest.Author
 		// todo: probably should be two different ones...
 		return aa.authorizer, nil
 	}
+	env, err := aa.getAzureEnvironment()
+	if err != nil {
+		return nil, err
```

Comment:
please cover this one with test - e.g. call `aa.getClientSecretAuthorizer` with invalid azure env name